### PR TITLE
Incorporate ShellCheck into GitHub Action

### PR DESCRIPTION
**Pull request by dtrodrigues**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
This will run ShellCheck on `bin/brew` and scripts ending in `.sh` not in a tap or in vendor.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -167,6 +167,14 @@ jobs:
         brew install vale
         vale docs/
 
+    - name: Run ShellCheck
+      run: |
+        brew install shellcheck
+        shellcheck --color bin/brew
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml:173]: This is (or at least should be) already run by `brew style`: https://github.com/Homebrew/brew/blob/081025c96a3fbd7210200f431c5110d05dd6d14c/Library/Homebrew/style.rb#L105-L115

**Comment by dtrodrigues** [comment]
It looks like https://github.com/Homebrew/brew/commit/892d0ac5febb943e103e01c863fc934c5663b8ee from April 2019 may have prevented shellcheck from running since then since the current GitHub action runs `brew style --display-cop-names`. The 3 detected errors were all introduced after that point (https://github.com/Homebrew/brew/commit/19fd56468fa5f27d4245ab1174973c62409157d3 https://github.com/Homebrew/brew/commit/99e308753b9d5c501a92fc225c2eca1ba634d298 https://github.com/Homebrew/brew/commit/5d3456f4856b3aa2606afd5fe3050293d6732c6b).

Also, I think it would be better to add a `#!/bin/bash` shebang to the `.sh` files missing them and remove the `--shell=bash` argument so the files are more explicit.

**Comment by MikeMcQuaid** [comment]
> It looks like [892d0ac](https://github.com/Homebrew/brew/commit/892d0ac5febb943e103e01c863fc934c5663b8ee) from April 2019 may have prevented shellcheck from running since then since the current GitHub action runs `brew style --display-cop-names`. The 3 detected errors were all introduced after that point ([19fd564](https://github.com/Homebrew/brew/commit/19fd56468fa5f27d4245ab1174973c62409157d3) [99e3087](https://github.com/Homebrew/brew/commit/99e308753b9d5c501a92fc225c2eca1ba634d298) [5d3456f](https://github.com/Homebrew/brew/commit/5d3456f4856b3aa2606afd5fe3050293d6732c6b)).

Good catch, I've re-enabled in https://github.com/Homebrew/brew/pull/7930.

> Also, I think it would be better to add a `#!/bin/bash` shebang to the `.sh` files missing them and remove the `--shell=bash` argument so the files are more explicit.

I'd rather not; these files are not designed to be executed as shell scripts but `source`d from other files.

**Comment by MikeMcQuaid** [comment]
Thanks for the PR @dtrodrigues, wouldn't have caught this without you ❤️

**Pull request closed without merging**